### ISSUE TITLE
AppsService: getInstalledAppForTeam - Invalid teamId and/or appId specified

### ISSUE THREAD

**Issue by ftntming** [MicrosoftDocs/msteams-docs#1738]
Can not save my connector, message is,

 Unable to save “abc” connector configuration. Please try again. 

Firefox console shows,

2020-06-05T05:44:42.193Z ngClickDecorator: value not found in enum type. value: 

2020-06-05T05:44:42.306Z Received error from connectors {"seq":1591335686918,"timestamp":1591335882300,"flightSettings":{"Name":"ConnectorFrontEndSettings","AriaSDKToken":"d127f72a3abd41c9b9dd94faca947689-d58285e6-3a68-4cab-a458-37b9d9761d35-7033","SPAEnabled":true,"ClassificationFilterEnabled":true,"ClientRoutingEnabled":true,"EnableYammerGroupOption":true,"EnableFadeMessage":false,"EnableDomainBasedOwaConnectorList":false,"EnableDomainBasedTeamsConnectorList":false,"DevPortalSPAEnabled":true,"ShowHomeNavigationButtonOnConfigurationPage":false,"DisableConnectToO365InlineDeleteFeedbackPage":true},"status":500,"clientType":"SkypeSpaces","connectorType":"30cb42b0-f92b-45c3-8abc-06187b92ce71","name":"handleMessageError"}

**Comment by natwallbank** [comment]
Any news on this? I'm seeing the same issue in a new connector I've been developing, following the example at: https://github.com/OfficeDev/microsoft-teams-sample-connector-csharp

I can enable the Save button via `microsoftTeams.settings.setValidityState(true)`, calls to `setSettings` and `getSettings` work fine, and I can retrieve the webhookUrl via the latter.  However, any requests to this URL result in 410 - Gone responses, which I'm guessing are related to this error that I see in the console prior to making any of these calls?

**Comment by ftntming** [comment]
Hi Nat Wallbank,

My issue was because I have an invalid "contentUrl" while trying to save
the settings. the following "undefined" was due to an variable not defined,
thus cause the
connector not saved properly.

const settings: any = {

    entityId: *"Some entity ID"*,
    contentUrl:
`https://teamsservice.mycompany.com/*undefined*/connector/config.html`,
    configName: *"Config Name"*
};
microsoftTeams.settings.setSettings(settings);

**Comment by natwallbank** [comment]
Thanks, I figured it out in the end but don't have a clear answer. In my case, I recreated the manifest from scratch using App Studio. Despite being identical, with the exception of the ID (though I did try regenerating that with the old manifest, to no avail) it now works without issue.

I can only assume that there was some state-related issue somewhere pertaining to the old configuration

**Comment by Wajeed-msft** [comment]
@natwallbank, @ftntming - Are you still facing this issue?

**Comment by natwallbank** [comment]
@Wajeed-msft I'm not, but then I ended up switching focus and developing a bot in preference to a connector.  

I did face a similar issue whenever I changed the messaging endpoint, but discovered that I needed to ensure I was returning an HTTP 200 for an OPTIONS request to that endpoint, that I assume is verified/validated on change of that URL (despite the UI not indicating that there was a problem).  I'm not sure if those two things are related.

Since then, I've had no further issues.
